packages/client/src/__tests__/integration/happy/exhaustive-schema-mongo/__snapshots__/test.ts.snap

```diff
@@ -4033,18 +4179,6 @@ export namespace Prisma {
   }
 
 
-  /**
-   * User: findUniqueOrThrow
-   */
-  export type UserFindUniqueOrThrowArgs = UserFindUniqueArgsBase
```

Comment:
Question: Are `UserFindUniqueArgsBase` still being generated? If so could we remove them?

Reply:
They are being generated and used for [extending the type to deprecate `rejectOnNotFound`](https://github.com/prisma/prisma/blob/b6decd849ccf0a02778283813ecf10968d661f65/packages/client/src/generation/TSClient/Args.ts#L111-L123)

I'm not completely sure how to monomorphise the base type and remove the child definition, as it's the only one. I'm going to dig a little bit and see what I find.

Reply:
Now tests are breaking because `ModelFind(First|Unique)Args` are duplicated. They  are generated once from the original type, and then a second time add the rejectOnNotFound option. 

We can do two things in here:
1. We could generate only one type, having the optional `rejectOnNotFound` property for both `Find(First|Unique)`  `Find(First|Unique)OrThrow` 
2. We could revert the changes and leave the aliased type

My understanding is that 1 will introduce an unexpected change. Before this, findUnique or findFirst, wouldn't allow `rejectOnNotFound` anymore.  It's only OrThrow variants the ones that do, by doing 1, we will be promoting to all the input types the rejectOnNotFound property. -- Is that maybe a bug? in any case it doesn't make sense to take back a backward compatibility that once was broken --

Reply:
Given the above, I think the best course of action as of now is to do 2. and revert the changes and leave the aliased type. And I did that in https://github.com/prisma/prisma/pull/16162/commits/3f9eedb0934050886d9403149a5c2cf7ebc11a04 and https://github.com/prisma/prisma/pull/16162/commits/c55b6e74f71f6293bfa77c78db021bc335fe2512

I will remove the revert and reverted commits before merge.